src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/caseupdated/CaseUpdatedAboutToSubmitHandler.java

```diff
@@ -206,11 +213,67 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
         if (sscsCaseData.isIbcCase()) {
             SscsUtil.setListAssistRoutes(sscsCaseData);
         }
+        updateOverrideFields(sscsCaseData, caseDetailsBefore);
         sscsCaseData.setPanelMemberComposition(panelCompositionService
                 .resetPanelCompositionIfStale(sscsCaseData, caseDetailsBefore));
         return preSubmitCallbackResponse;
     }
 
+    private void updateOverrideFields(SscsCaseData sscsCaseData, Optional<CaseDetails<SscsCaseData>> caseDetailsBefore) {
+        if (isNull(sscsCaseData.getSchedulingAndListingFields().getDefaultListingValues()) || caseDetailsBefore.isEmpty()) {
+            return;
+        }
+        OverrideFields overrideFields = ofNullable(sscsCaseData.getSchedulingAndListingFields().getOverrideFields())
+                .orElse(OverrideFields.builder().build());
+        HearingOptions hearingOptions = sscsCaseData.getAppeal().getHearingOptions();
+        if (nonNull(hearingOptions)) {
+            if (isYes(hearingOptions.getLanguageInterpreter()) && isNull(hearingOptions.getLanguages())) {
+                return;
+            }
+            if (hasInterpreterChanged(sscsCaseData, caseDetailsBefore.get())) {
+                updateOverrideInterpreter(hearingOptions, overrideFields);
+                updateOverrideDuration(sscsCaseData, overrideFields);
+            } else if (hasLanguageChanged(sscsCaseData, caseDetailsBefore.get())) {
+                updateOverrideInterpreter(hearingOptions, overrideFields);
+            }
+            sscsCaseData.getSchedulingAndListingFields().setOverrideFields(overrideFields);
```

Comment:
it would be better to return the override fields and update them in the main handle method, rather than mutate the caseData object here